### ISSUE TITLE
Filmon plugin SSLError issue

### ISSUE THREAD

**Issue by mtsekowski** [streamlink/streamlink#2784]
## Issue

- [ ] This is not a bug report, feature request, or plugin issue/request.
- [x] I have read #the contribution guidelines.

### Description

I have the latest 1.3.1 streamlink version installed on Raspbian Buster linux.
It looks like FilmOn has recently added http to https redirect resulting in the following SSLError("bad handshake). 
Changing source url to **https**://www.filmon.com/tv/bbc-one for example results in the same error
Also tried to add --http-no-ssl-verify parameter. It didn't help.
### Expected / Actual behavior

The stream doesn't work

### Reproduction steps / Explicit stream URLs to test

1. Run "streamlink http://www.filmon.com/tv/bbc-one" 

### Logs

```
Unable to open URL: http://www.filmon.com/tv/bbc-one (HTTPSConnectionPool(host='www.filmon.com', port=443): Max retries exceeded with url: /tv/bbc-one (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'tls12_check_peer_sigalg', 'wrong signature type')],)",),)))

```

### Additional comments, screenshots, etc.

[Love Streamlink? Please consider supporting our collective. Thanks!](https://opencollective.com/streamlink/donate)

**Comment by johnthecracker** [comment]
works for me, streamlink-1.3.1-20200129.exe

try: --http-no-ssl-verify

```
$url = ClipGet()
run('streamlink.exe --http-no-ssl-verify --player=mpv.exe "' & $url & '" best')
```

```
[cli][info] Found matching plugin filmon for URL http://www.filmon.com/tv/bbc-one
[cli][info] Available streams: low (worst), high (best)
[cli][info] Opening stream: high (hls-filmon)
[cli][info] Starting player: mpv.exe
```

**Comment by mkbloke** [comment]
Interestingly, I do not get an SSL error, but I also do not seem to get BBC One using any of the below URLs that I've tested:
```
streamlink filmon.com/tv/bbc-one low
streamlink filmon.com/tv/bbc-one high
streamlink http://www.filmon.com/tv/bbc-one high
streamlink https://www.filmon.com/tv/bbc-one high
```
I updated to latest master before trying the above.  I'm not sure what channel is actually streaming, but it's not BBC One.

Using the previous URL path:
```
streamlink filmon.com/channel/bbc-one low
```

Also results in the same non-BBC One channel, but also no SSL error.

**Comment by mkbloke** [comment]
Scrap the above, there was another issue on my system that seemed to cause the behaviour I observed before.  I tried removing:
```
~/.cache/streamlink/plugin-cache.json
```

After that, I'm getting BBC One again, with no SSL errors.

**Comment by mtsekowski** [comment]
Thank you for the response,
I have finally fixed the problem. It turned out it was my system's SSL configuration issue instead of streamlink issue. The protocol version must have changed after a system update.
I have no longer the issue after replacing **/etc/ssl/openssl.cnf** file:
```
[system_default_sect]
MinProtocol = TLSv1.2
```
by
```
[system_default_sect]
MinProtocol = TLSv1.
```
Here is where I found the solution from: https://git.weboob.org/weboob/weboob/issues/217
Thank you guys!
